corva/models/stream.py

```diff
@@ -28,24 +23,44 @@ class Record(BaseEventData):
     data: RecordData
 
 
-class AppMetadata(BaseEventData):
+class AppMetadata(CorvaBaseModel):
     app_connection_id: int
     app_version: Optional[int] = None
 
 
-class StreamEventMetadata(BaseEventData):
+class StreamEventMetadata(CorvaBaseModel):
     app_stream_id: int
+    source_type: Optional[str] = None
     apps: Dict[str, AppMetadata]
 
 
-class StreamEventData(BaseEventData):
+class StreamEventData(CorvaBaseModel):
     app_key: Optional[str] = None
     records: List[Record]
     metadata: StreamEventMetadata
+    asset_id: int = None
```

Comment:
Optional[int]?

Reply:
It is required to be int. the reason for such type hint: https://pydantic-docs.helpmanual.io/usage/validators/#validate-always
added comment to code